File: 387. First Unique Character in a String.cpp
#include <string>
#include <iostream>
#include <algorithm>
using namespace std;
class Solution {
public:
    int firstUniqChar(string s1234556) {
         char ret= rec(s1234556);
         if(ret !='.') return s1234556.find(ret);
         return  -1;

    }

    char rec(string s1){
        
        if (s1 == "" ){
            return '.';
        } else if (s1.size()==1) return s1[0];

        char peek = s1[0];
        string rest =  s1.substr(1);

        if (rest.find(peek)==string::npos){
            return peek;
        }
        else   {
           rest.erase(std::remove(rest.begin(), rest.end(), peek), rest.end());
            return rec(rest);
        }
    }
};

File: 387. First Unique Character in a String.py
class Solution:
    def firstUniqChar(self, string111: str) -> int:
        def rec(s1):
            #print(s1)
            if s1 is None or s1 =='': return None
            if len(s1) == 1: return s1

            peek = s1[0]
            rest = s1[1:]

            if peek not in rest: return peek
            return rec(rest.replace(peek,''))

        ret_char = rec(string111)

        if ret_char: return string111.find(ret_char)
        return -1

print(Solution().firstUniqChar("loveleetcode"))
